packages/spruce-skill-server/gql/Schema.ts

```diff
@@ -1,116 +1,214 @@
-/* eslint-disable @typescript-eslint/no-var-requires */
 import globby from 'globby'
-import Debug from 'debug'
-import { GraphQLObjectType, GraphQLSchema, GraphQLSchemaConfig } from 'graphql'
+import {
+	GraphQLObjectType,
+	GraphQLSchema,
+	GraphQLSchemaConfig,
+	parse,
+	extendSchema
+} from 'graphql'
+
+import { addResolveFunctionsToSchema } from 'graphql-tools/dist/generate'
+
+import fs from 'fs'
+
 import helpers from './helpers'
 import { ISpruceContext } from '../interfaces/ctx'
-const debug = Debug('spruce-skill-server')
+import config from 'config'
 
 export default class Schema {
+	public readonly gqlSchema: GraphQLSchema
+
 	public constructor(options: { ctx: ISpruceContext; gqlDir: string }) {
 		const { ctx, gqlDir } = options
-		try {
-			ctx.gql = {
-				helpers: helpers(ctx),
-				// @ts-ignore
-				types: {}
+		ctx.gql = {
+			helpers: helpers(ctx),
+			// @ts-ignore
+			types: {}
+		}
+		const coreTypePaths = globby.sync([
+			`${__dirname}/types/**/!(index|types|_helpers).(ts|js|gql)`
+		])
+		const typePaths = globby.sync([
+			`${gqlDir}/types/**/!(index|types|_helpers).(ts|js|gql)`
+		])
+		const resolverPaths = globby.sync([
+			`${gqlDir}/resolvers/**/!(index|types|_helpers).(ts|js)`
+		])
+		let queries = {}
+		let mutations = {}
+		let subscriptions = {}
+
+		let sdl = ``
+		let allResolvers = {
+			Query: {},
+			Mutation: {}
+		}
+
+		// Load GQL types first and assign to ctx.gql.types[<type name>]
+		const allTypePaths = [...coreTypePaths, ...typePaths]
+		allTypePaths.forEach(path => {
+			log.debug(`checking GQL type @ ${path}`)
+			if (path.search('.d.ts') > -1) {
```

Comment:
Did the globby we had before not work?  It _should_ have already been excluding `.d.ts` files